src/Domain/Campaign.php

```diff
@@ -243,22 +260,43 @@ public function isReady(): bool
         return $this->ready;
     }
 
-    public function updateFromSfPull(
+    final public function updateFromSfPull(
```

Comment:
has to be final for Psalm to be satisfied that this can reliably take on some of the constructor's job of setting fields that don't have default values.